schema/deploy/shipping/views.sql

```diff
@@ -593,3 +592,73 @@ create or replace view shipping.metadata_for_augur_build_v3 as
 
 comment on view shipping.metadata_for_augur_build_v3 is
 		'View of metadata necessary for SFS augur build';
+
+
+create or replace view shipping.sample_with_best_available_site_details_v1 as
+
+    with specimen_manifest_site_data as (
+        select distinct on (manifest_id)
+            sample_id,
+            trim(both ' ' from document->>'swab_site') as swab_site,
+            trim(both ' ' from document->>'sample_origin') as sample_origin
+        from
+            receiving.manifest,
+            jsonb_to_recordset(processing_log) as log(sample_id integer, timestamp date)
+        where sample_id is not null
+        order by manifest_id, timestamp desc
+    ),
+
+    site_details as (
+      select
+          site_id,
+          site.details->>'type' as site_type,
+          site.details->>'category' as site_category,
+          coalesce(site.details->>'swab_site', site.details->>'sample_origin') as manifest_regex
+      from warehouse.site
+    ),
+
+    samples_with_manifest_site_data as (
+      select
+        sample_id,
+        site_id,
+
+        coalesce(
+          case
+              -- Environmental samples must be processed first, because they're
+              -- often taken at existing human swab sites
+              when manifest.sample_origin = 'es'
+                then 'environmental'
+              else manifest.swab_site
+          end,
+
+          manifest.sample_origin
+        ) as site_manifest_details,
+
+        case
+          when site_id is null
+            then 'f'
+          else 't'
+        end as has_encounter_data
```

Comment:
It's slightly weird to me that `has_encounter_data` is text and not just a boolean.

Reply:
Agreed. Good catch. 
I was thinking that because boolean values in Postgres can be matched with ` = 't'` or ` = 'f'` that they could also be inserted as such and convert to boolean. I was mistaken. I just changed this commit to use `FALSE` and `THEN`, and now the `pg_typeof()` of this column is boolean as expected.

Reply:
FYI, if all you want is the boolean value, this can be shortened to 
`site_id is not null as has_encounter_data`

Reply:
Thanks for the tip! Fixed up.